src/wallet/mnemoniccontainer.h

```diff
@@ -0,0 +1,47 @@
+#ifndef ZCOIN_HDCHAIN_H
+#define ZCOIN_HDCHAIN_H
+
+#include "support/allocators/secure.h"
+#include "uint256.h"
+#include "serialize.h"
+
+class MnemonicConatiner
+{
+public:
+    SecureVector seed;
+    SecureVector mnemonic;
+    bool fIsCrypted;
+    bool f12Words;
+
+    MnemonicConatiner() { SetNull(); }
+
+    ADD_SERIALIZE_METHODS;
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion)
+    {
+            READWRITE(mnemonic);
+            READWRITE(seed);
```

Comment:
Why do we need to save seed? The only purpose of `MnemonicContainer` is to save mnemonic for later recovery by user. Seed is saved within `wallet.dat` by bitcoin code, no need to duplicate it here.

Reply:
It only being kept here.

Reply:
Keeping secret data when unnecessary is generally not a good idea. And seed is essentially ultimate attacker's goal

Reply:
I got your point but we are keeping seed here for deriving keys, please take a look on GenerateNewKey() and GetKeyFromKeypath() functions.

Reply:
I still don't get it. When initializing wallet with mnemonic we generate new seed and from that moment it is in the hands of bitcoin code. Why do we need to do our own way of generating new key? Why can't bitcoin code generate one for us?

Reply:
Mnemonic seed is being generated 64 byte, but Key we can set only 32, and using other hacks is generating addresses not compatible to hardware wallets.